docs/kb/function-tutorial-intro.md

```diff
@@ -0,0 +1,239 @@
+
+## The anatomy of a VEBA function
+All VEBA functions all have the same relative format regardless of what programming language the functions are written in.  Lets look at the [kn-ps-slack](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powershell/kn-ps-slack) function.
+
+<img src="./img/kn-ps-slack-directory.png" width="100%" align="center" class="border m-1 p-1"/>
+
+
+- **test** - this directory contains scripts to test and run the VEBA function locally on your workstation, without need for the VEBA appliance.
+- **Dockerfile** contains the instructions for the Docker application on how to build the Docker image.  You most likely will not need to modify this file.
+- **README.md** is the default web page you see at [kn-ps-slack](https://github.com/vmware-samples/vcenter-event-broker-appliance/tree/master/examples/knative/powershell/kn-ps-slack).  It is written in markdown language.
+- **function.yaml** contains the instructions to load the VEBA function into the Kubernetes cluster.  This file has two pieces of information that you may want to change:
+  1. The vCenter Event that triggers the function.  The default event is: **VmPoweredOffEvent** but this may be modified to suit your requirements.
+  2. The location of the Docker image to load into Kubernetes.  If the PowerShell code will not be modified, you can leave the default image as is.  If you modify the PowerShell code in the handler file, you will need to build a new Docker image and specify the name/location in this file.
+- **handler.ps1** contains the PowerShell code that runs when the specified vCenter event triggers it.  The handler.ps1 file is injected into the Docker image which must be rebuilt every time the code changes.
+- **slack_secret.json** contains the Slack key for your Slack channel.  This file will be used to create a Kubernetes secret that the function will read.  Most functions will include a secret file that contains: usernames/passwords, keys, or other sensitive information that will be used to create a Kubernetes secret.
+
+**What are Kubernetes secrets?**  The VEBA appliance runs a Kubernetes cluster on top of the Photon OS.  This allows the VEBA functions to be packaged as containers that are then deployed into the Kubernetes cluster.  A Kubernetes secret is an object that contains a small amount of sensitive data that can be read by a Kubernets pod (pods are made up of containers).  Secrets allow sensitive or confidential data to be packaged separate from a container's application code.
+
+Also note the link in the README for the available video tutorial which may be of help: [video tutorial](https://youtu.be/vLyeZrm0jk4)
```

Comment:
nit: I'd remove this as it might distract the reader from this tutorial. Alternatively, you can reword this sentence because it wasn't immediately clear what the link does before I clicked.